app/scripts/modules/core/src/function/CreateFunctionButton.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+
+import { Application } from 'core/application';
+import { CloudProviderRegistry, ProviderSelectionService } from 'core/cloudProvider';
+import { IFunction } from 'core/domain';
+import { IFunctionUpsertCommand } from 'core/function';
+import { ReactInjector } from 'core/reactShims';
+import { IModalComponentProps, Tooltip } from 'core/presentation';
+
+export interface IFunctionModalProps extends IModalComponentProps {
+  className?: string;
+  dialogClassName?: string;
+  app: Application;
+  forPipelineConfig?: boolean;
+  functionDef: IFunction;
+  command?: IFunctionUpsertCommand; // optional, when ejecting from a wizard
+  closeModal?: (functionCommand: IFunctionUpsertCommand) => void; // provided by ReactModal
+  dismissModal?: (rejectReason?: any) => void; // provided by ReactModal
+}
+
+export interface ICreateFunctionButtonProps {
+  app: Application;
+}
+
+export class CreateFunctionButton extends React.Component<ICreateFunctionButtonProps> {
+  constructor(props: ICreateFunctionButtonProps) {
```

Comment:
remove empty constructor

Reply:
Done. change will appear in the next push.